consensus/obcpbft/obc-batch.go

```diff
@@ -90,20 +92,37 @@ func newObcBatch(id uint64, config *viper.Viper, stack consensus.Stack) *obcBatc
 	op.custodyTimerChan = make(chan custodyInfo)
 	op.execChan = make(chan *execInfo)
 
-	op.complainer = newComplainer(op, op.pbft.requestTimeout, op.pbft.requestTimeout)
-	op.deduplicator = newDeduplicator()
-
 	// create non-running timer
 	op.batchTimer = time.NewTimer(100 * time.Hour) // XXX ugly
 	op.batchTimer.Stop()
 
 	op.idleChan = make(chan struct{})
 	op.viewChanged = make(chan struct{})
 
+	go op.waitForID(config)
```

Comment:
why does this go here, and not at the very top of the constructor, and have the rest of the setup be performed later?